service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -58,8 +58,18 @@ readonly TERRA_SERVER=$(get_metadata_value "instance/attributes/terra-cli-server
 if [[ -n "${TERRA_SERVER}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra server set --name=${TERRA_SERVER}"
 fi
+
 # Set the CLI terra workspace id using the VM metadata, if set.
 readonly TERRA_WORKSPACE=$(get_metadata_value "instance/attributes/terra-workspace-id")
 if [[ -n "${TERRA_WORKSPACE}" ]]; then
   sudo -u "${JUPYTER_USER}" sh -c "terra workspace set --id=${TERRA_WORKSPACE} --defer-login"
 fi
+
+# Log in with app-default-credentials
+sudo -u "${JUPYTER_USER}" sh -c "terra auth login --mode=APP_DEFAULT_CREDENTIALS"
+
+# Get all the git repo clone urls from the workspace resource list with type GIT_REPO and try to clone all of them.
+sudo -u "${JUPYTER_USER}" sh -c "terra resource list --type=GIT_REPO --format=JSON" | jq -r '.[] | {gitRepoUrl} | join (" ") ' | while read -r line; do
```

Comment:
I think it's possible to have two reference resources pointing to the same github repo. `jq` has a way to get the unique entries in a list using `unique` or `unique_by`. If you can't create a clashing reference, then it's not a problem.

Reply:
It's definitely possible to create two references to the same underlying object, so `git clone` may complain here

Reply:
good point. added unique check.